src/Common/StackTrace.cpp

```diff
@@ -267,12 +272,45 @@ void StackTrace::forEachFrame(
 #else
     UNUSED(fatal);
 
+    auto split_by_whitespace = [](const std::string & str)
+    {
+        std::vector<std::string> splitted;
+        ssize_t prev = -1;
+        for (size_t pos = 0; pos < str.size(); ++pos)
+        {
+            if (!isWhitespaceASCII(str[pos]))
+            {
+                if (prev == -1)
+                    prev = pos;
+                continue;
+            }
+
+            if (prev != -1)
+            {
+                splitted.push_back(str.substr(prev, pos - prev));
+                prev = -1;
+            }
+        }
+
+        if (prev != -1)
+            splitted.push_back(str.substr(prev, str.size() - prev));
+
+        return splitted;
+    };
+
+    char** strs = ::backtrace_symbols(frame_pointers.data(), static_cast<int>(size));
+
     for (size_t i = offset; i < size; ++i)
     {
         StackTrace::Frame current_frame;
         current_frame.virtual_addr = frame_pointers[i];
+        auto splitted = split_by_whitespace(strs[i]);
+        assert(splitted.size() >= 6);
```

Comment:
Missing comments about "6".

Reply:
I wrote the comment, but will also paste it here. The `backtrace_symbols` returns you an array of strings in format like below. The symbol name is mangled and we need to split everything to be able to demangle it. 
```
(lldb) run local -q "SELECT throwIf(1)"
Process 9100 launched: '/Users/nikitamikhaylov/ClickHouse/ClickHouse/build/programs/clickhouse' (arm64)
0   clickhouse                          0x0000000100028024 _ZNSt9exception7captureEv + 36
1   clickhouse                          0x0000000100027ff0 _ZNSt9exceptionC2B6v15000Ev + 48
2   clickhouse                          0x0000000118c11144 _ZN4Poco9ExceptionC2ERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEEi + 36
3   clickhouse                          0x000000010843663c _ZN2DB9ExceptionC2EONS0_13MessageMaskedEib + 60
4   clickhouse                          0x00000001084367a0 _ZN2DB9ExceptionC1EONS0_13MessageMaskedEib + 64
5   clickhouse                          0x0000000100013434 _ZN2DB9ExceptionC1EONSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEEib + 84
6   clickhouse                          0x000000010001330c _ZN2DB9ExceptionC1IJNSt3__112basic_stringIcNS2_11char_traitsIcEENS2_9allocatorIcEEEEEEEi22FormatStringHelperImplIJDpNS2_13type_identityIT_E4typeEEEDpOSB_ + 300
7   clickhouse                          0x0000000106c5cd08 _ZNK2DB12_GLOBAL__N_115FunctionThrowIf7executeIDuEEN3COWINS_7IColumnEE13immutable_ptrIS4_EEPKS4_RKNSt3__18optionalINSA_12basic_stringIcNSA_11char_traitsIcEENSA_9allocatorIcEEEEEERKNSB_IiEE + 456
8   clickhouse                          0x0000000106c5bb98 _ZNK2DB12_GLOBAL__N_115FunctionThrowIf11executeImplERKNSt3__16vectorINS_21ColumnWithTypeAndNameENS2_9allocatorIS4_EEEERKNS2_10shared_ptrIKNS_9IDataTypeEEEm + 1272
9   clickhouse                          0x000000010001654c _ZNK2DB35FunctionToExecutableFunctionAdaptor11executeImplERKNSt3__16vectorINS_21ColumnWithTypeAndNameENS1_9allocatorIS3_EEEERKNS1_10shared_ptrIKNS_9IDataTypeEEEm + 76
10  clickhouse                          0x000000010f44fb94 _ZNK2DB19IExecutableFunction35executeWithoutLowCardinalityColumnsERKNSt3__16vectorINS_21ColumnWithTypeAndNameENS1_9allocatorIS3_EEEERKNS1_10shared_ptrIKNS_9IDataTypeEEEmb + 628
11  clickhouse                          0x000000010f450ae8 _ZNK2DB19IExecutableFunction27executeWithoutSparseColumnsERKNSt3__16vectorINS_21ColumnWithTypeAndNameENS1_9allocatorIS3_EEEERKNS1_10shared_ptrIKNS_9IDataTypeEEEmb + 1640
12  clickhouse                          0x000000010f451ac8 _ZNK2DB19IExecutableFunction7executeERKNSt3__16vectorINS_21ColumnWithTypeAndNameENS1_9allocatorIS3_EEEERKNS1_10shared_ptrIKNS_9IDataTypeEEEmb + 1448
13  clickhouse                          0x00000001124f80b0 _ZN2DBL13executeActionERKNS_17ExpressionActions6ActionERNS_12_GLOBAL__N_116ExecutionContextEb + 1232
14  clickhouse                          0x00000001124f7310 _ZNK2DB17ExpressionActions7executeERNS_5BlockERmb + 720
15  clickhouse                          0x0000000115ffb4a8 _ZN2DB19ExpressionTransform9transformERNS_5ChunkE + 136
16  clickhouse                          0x000000010bf5e9ec _ZN2DB16ISimpleTransform9transformERNS_5ChunkES2_ + 44
17  clickhouse                          0x0000000115917624 _ZN2DB16ISimpleTransform4workEv + 100
18  clickhouse                          0x0000000115958d2c _ZN2DBL10executeJobEPNS_14ExecutingGraph4NodeEPNS_20ReadProgressCallbackE + 44
19  clickhouse                          0x0000000115958a5c _ZN2DB22ExecutionThreadContext11executeTaskEv + 284
20  clickhouse                          0x00000001159624f0 _ZN2DB16PipelineExecutor15executeStepImplEmPNSt3__16atomicIbEE + 336
21  clickhouse                          0x000000011596296c _ZN2DB16PipelineExecutor19executeSingleThreadEm + 44
22  clickhouse                          0x00000001159615c8 _ZN2DB16PipelineExecutor11executeImplEmb + 232
23  clickhouse                          0x0000000115960d70 _ZN2DB16PipelineExecutor7executeEmb + 176
24  clickhouse                          0x000000011597a4e8 _ZN2DBL14threadFunctionERNS_28PullingAsyncPipelineExecutor4DataENSt3__110shared_ptrINS_11ThreadGroupEEEmb + 296
25  clickhouse                          0x000000011597a37c _ZZN2DB28PullingAsyncPipelineExecutor4pullERNS_5ChunkEyENK3$_0clEv + 124
26  clickhouse                          0x000000011597a2d8 _ZNSt3__18__invokeB6v15000IRZN2DB28PullingAsyncPipelineExecutor4pullERNS1_5ChunkEyE3$_0JEEEDTclclsr3stdE7declvalIT_EEspclsr3stdE7declvalIT0_EEEEOS7_DpOS8_ + 24
27  clickhouse                          0x000000011597a29c _ZNSt3__118__apply_tuple_implB6v15000IRZN2DB28PullingAsyncPipelineExecutor4pullERNS1_5ChunkEyE3$_0RNS_5tupleIJEEEJEEEDcOT_OT0_NS_15__tuple_indicesIJXspT1_EEEE + 28
28  clickhouse                          0x000000011597a1a0 _ZNSt3__15applyB6v15000IRZN2DB28PullingAsyncPipelineExecutor4pullERNS1_5ChunkEyE3$_0RNS_5tupleIJEEEEEDcOT_OT0_ + 32
29  clickhouse                          0x000000011597a0ac _ZZN24ThreadFromGlobalPoolImplILb1EEC1IZN2DB28PullingAsyncPipelineExecutor4pullERNS2_5ChunkEyE3$_0JEEEOT_DpOT0_ENUlvE_clEv + 140
30  clickhouse                          0x0000000115979ff8 _ZNSt3__18__invokeB6v15000IRZN24ThreadFromGlobalPoolImplILb1EEC1IZN2DB28PullingAsyncPipelineExecutor4pullERNS4_5ChunkEyE3$_0JEEEOT_DpOT0_EUlvE_JEEEDTclclsr3stdE7declvalIS9_EEspclsr3stdE7declvalISB_EEEESA_SD_ + 24
31  clickhouse                          0x0000000115979fb8 _ZNSt3__128__invoke_void_return_wrapperIvLb1EE6__callIJRZN24ThreadFromGlobalPoolImplILb1EEC1IZN2DB28PullingAsyncPipelineExecutor4pullERNS6_5ChunkEyE3$_0JEEEOT_DpOT0_EUlvE_EEEvDpOT_ + 24
Code: 395. DB::Exception: Value passed to 'throwIf' function is non-zero: while executing 'FUNCTION throwIf(1 :: 0) -> throwIf(1) UInt8 : 1'. (FUNCTION_THROW_IF_VALUE_IS_NON_ZERO)
```